tools/generate-sponsors.ts

```diff
@@ -4,90 +4,198 @@ import * as path from 'path';
 
 const graphqlEndpoint = 'https://api.opencollective.com/graphql/v2';
 
-const graphqlQuery = `{
-  account(slug: "typescript-eslint") {
-    orders(status: ACTIVE, limit: 1000) {
-      totalCount
-      nodes {
-        tier {
-          slug
+const queries = {
+  account: `{
+    account(slug: "typescript-eslint") {
+      orders(status: ACTIVE, limit: 1000) {
+        totalCount
+        nodes {
+          tier {
+            slug
+          }
+          fromAccount {
+            id
+            imageUrl
+            name
+            website
+          }
         }
-        fromAccount {
-          id
-          name
-          slug
-          website
-          imageUrl
-          description
+      }
+    }
+  }`,
+  collective: `{
+    collective(slug: "typescript-eslint") {
+      members(limit: 1000, role: BACKER) {
+        nodes {
+          account {
+            id
+            imageUrl
+            name
+            website
+          }
+          tier {
+            amount {
+              valueInCents
+            }
+            orders(limit: 100) {
+              nodes {
+                amount {
+                  valueInCents
+                }
+              }
+            }
+          }
+          totalDonations {
+            valueInCents
+          }
+          updatedAt
         }
       }
     }
-  }
-}`;
+  }`,
+};
 
-const excludedIds = new Set([
-  '53kzxy4v-07wlr6mr-o9epmj9n-o8agdbe5', // Josh Goldberg
-]);
-
-interface SponsorsData {
-  data: {
-    account: {
-      orders: {
-        nodes: SponsorNode[];
-      };
-    };
+interface AccountData {
+  orders: {
+    nodes: OrderNode[];
   };
 }
 
-interface SponsorNode {
-  fromAccount: {
-    description: string;
-    id: string;
-    imageUrl: string;
-    name: string;
-    slug: string;
-    twitterHandle: string;
-    website: string;
+interface OrderNode {
+  fromAccount: MemberAccount;
+  tier?: Tier;
+}
+
+interface Tier {
+  slug: string;
+}
+
+interface CollectiveData {
+  members: {
+    nodes: MemberNode[];
   };
-  tier?: {
-    slug: string;
+}
+
+interface MemberNode {
+  account: MemberAccount;
+  totalDonations: {
+    valueInCents: number;
   };
 }
 
-async function main(): Promise<void> {
+interface MemberAccount {
+  id: string;
+  imageUrl: string;
+  name: string;
+  twitterHandle: string;
+  website: string;
+}
+
+interface MemberAccountAndTier extends MemberAccount {
+  tier?: Tier;
+}
+
+const excludedNames = new Set([
+  'Guest', // Apparent anonymous donor equivalent without an avatar
+  'Josh Goldberg', // Team member 💖
+
+  // These names *seem* to be spam websites, but we're not sure.
+  // If your name is mistakenly on this list, we're sorry; please let us know!
+  'Deal Empire',
+  'Florian Studio',
+  'java',
+  'Loyalty Leo',
+  'Penalty.com',
```

Comment:
https://github.com/typescript-eslint/typescript-eslint/pull/4758#pullrequestreview-937046372

@bradzacher that one trailing box is really irritating, isn't it? 😄 

OpenCollective support said they'd stop the spam, but these accounts are still showing up.